zookeeper-client/zookeeper-client-c/src/cli.c

```diff
@@ -686,7 +686,7 @@ int main(int argc, char **argv) {
                   sizeof(cmd));
           return 2;
         }
-        strncpy(cmd, argv[2]+4, sizeof(cmd));
+        strncpy(cmd, argv[2]+4, sizeof(cmd) - 1);
```

Comment:
This seems a bit of a bogus "fix" specifically for the truncation warning. The change is basically saying that the src n is less than the max buffer size. This happens to be true here given the "cmdlen" check above, and the "+4" offset on the src. However it's a bit of a hash, no?

A google for "memcpy vs strncpy" shows similar debates, and I also see google results for "stringop-truncation" which results in links such as https://stackoverflow.com/questions/50198319/gcc-8-wstringop-truncation-what-is-the-good-practice

@eolivelli should we consider changing the idiom here? What do you think?